scvi/model/base/_pyromixin.py

```diff
@@ -0,0 +1,415 @@
+
+        if batch_size is None:
+            batch_size = self.adata.n_obs
+        train_dl = AnnDataLoader(self.adata, shuffle=False, batch_size=batch_size)
+        # sample local parameters
+        i = 0
+        for tensor_dict in track(
+            train_dl,
+            style="tqdm",
+            description="Sampling local variables, batch: ",
+        ):
+            args, kwargs = self.module._get_fn_args_from_batch(tensor_dict)
+            args = [a.to(device) for a in args]
+            kwargs = {k: v.to(device) for k, v in kwargs.items()}
+            self.to_device(device)
+
+            if i == 0:
+                sample_kwargs_obs_plate = sample_kwargs.copy()
+                sample_kwargs_obs_plate[
+                    "return_sites"
+                ] = self._check_obs_plate_return_sites(sample_kwargs)
+                sample_kwargs_obs_plate["show_progress"] = False
+                obs_plate = self._find_plate_dimension(args, kwargs)
+                obs_plate_dim = list(obs_plate.values())[0]
+                samples = self._get_posterior_samples(
+                    args, kwargs, **sample_kwargs_obs_plate
+                )
+            else:
+                samples_ = self._get_posterior_samples(
+                    args, kwargs, **sample_kwargs_obs_plate
+                )
+
+                samples = {
+                    k: np.array(
+                        [
+                            np.concatenate(
+                                [samples[k][j], samples_[k][j]],
+                                axis=obs_plate_dim,
+                            )
+                            for j in range(
+                                len(samples[k])
+                            )  # for each sample (in 0 dimension
+                        ]
+                    )
+                    for k in samples.keys()  # for each variable
+                }
+            i += 1
+
+        # sample global parameters
+        tensor_dict = next(iter(train_dl))
+        args, kwargs = self.module._get_fn_args_from_batch(tensor_dict)
+        args = [a.to(device) for a in args]
+        kwargs = {k: v.to(device) for k, v in kwargs.items()}
+        self.to_device(device)
+
+        global_samples = self._get_posterior_samples(args, kwargs, **sample_kwargs)
+        global_samples = {
+            k: v
+            for k, v in global_samples.items()
+            if k not in list(self.module.model.list_obs_plate_vars()["sites"].keys())
+        }
+
+        for k in global_samples.keys():
+            samples[k] = global_samples[k]
+
+        self.module.to(device)
+
+        return samples
+
+    def sample_posterior(
+        self,
+        num_samples: int = 1000,
+        return_sites: Optional[list] = None,
+        use_gpu: bool = False,
+        batch_size: int = 128,
+        sample_kwargs=None,
+        return_samples: bool = False,
+    ):
+        """
+        Generate samples from posterior distribution for each parameter
```

Comment:
for these docstrings there should be one line description with period and then a line break and an empty line and then you can have a paragraph